projectforum/projects/tests/test_project_filter.py

```diff
@@ -0,0 +1,186 @@
+        self.assertEqual(0, len(contents['projects']))
+
+    def test_list_projects_by_title_down(self):
+        test_create_projects.create_many_projects()
+        response = project_filters.get_projects(status=1,
+                                                keywords='',
+                                                order='title',
+                                                salary='lump',
+                                                ascending=False)
+        contents = response
+        self.assertEqual('F', contents[0].title)
+        self.assertEqual('A', contents[5].title)
+        self.assertEqual('C', contents[3].title)
+
+    def test_list_projects_by_title_up(self):
+        test_create_projects.create_many_projects()
+        response = project_filters.get_projects(status=1,
+                                                keywords='',
+                                                order='title',
+                                                salary='lump',
+                                                ascending=True)
+        contents = response
+        self.assertEqual('A', contents[0].title)
+        self.assertEqual('F', contents[5].title)
+        self.assertEqual('D', contents[3].title)
+
+    def test_list_projects_by_newest(self):
+        test_create_projects.create_many_projects()
+        response = project_filters.get_projects(status=1,
+                                                keywords='',
+                                                order='timestamp',
+                                                salary='lump',
+                                                ascending=False)
+        contents = response
+        self.assertEqual('B', contents[0].title)
+        self.assertEqual('E', contents[5].title)
+        self.assertEqual('D', contents[3].title)
+
+    def test_list_projects_by_oldest(self):
+        test_create_projects.create_many_projects()
+        response = project_filters.get_projects(status=1,
+                                                keywords='',
+                                                order='timestamp',
+                                                salary='lump',
+                                                ascending=True)
+        contents = response
+        self.assertEqual('E', contents[0].title)
+        self.assertEqual('B', contents[5].title)
+        self.assertEqual('C', contents[3].title)
+
+    def test_list_projects_by_pay_lump_down(self):
+        test_create_projects.create_many_projects()
+        response = project_filters.get_projects(status=1,
+                                                keywords='',
+                                                order='payment',
+                                                salary='lump',
+                                                ascending=False)
+        contents = response
+        self.assertEqual('D', contents[0].title)
+        self.assertEqual('B', contents[5].title)
```

Comment:
I'm not filtering with those variables.  I only filter by keyword.  If you order by payment, it will first order by your salary request (lump vs hourly).  So you get your list of hourly sorted top to bottom or whatnot.  It will then display the hourly or lump list afterwards.  So you get the full list.  Do you want it to be filtered specifically?  This is more of a user experience question here, so let me know what you prefer.  Am neutral to both.